fe/fe-core/src/main/java/com/starrocks/load/streamload/StreamLoadTask.java

```diff
@@ -997,8 +997,12 @@ public void afterCommitted(TransactionState txnState, boolean txnOperated) throw
         }
 
         // sync stream load collect profile
-        if (isSyncStreamLoad() && coord.isProfileAlreadyReported()) {
-            collectProfile();
+        if (isSyncStreamLoad() && coord.isEnableLoadProfile()) {
+            if (coord.isProfileAlreadyReported()) {
+                // collect profile only when be has reported
+                collectProfile();
+            }
+            // make sure query info is unregistered
             QeProcessorImpl.INSTANCE.unregisterQuery(loadId);
```

Comment:
Mv `QeProcessorImpl.INSTANCE.unregisterQuery(loadId);` outside of if?

Reply:
have moved it into `finally` block